app/controllers/api/migrations.php

```diff
@@ -0,0 +1,554 @@
+                'projectId' => $projectId,
+                'apiKey' => $apiKey,
+            ]),
+            'resources' => $resources,
+            'statusCounters' => '{}',
+            'resourceData' => "{}",
+            'errors' => []
+        ]));
+
+        $eventsInstance->setParam('migrationId', $migration->getId());
+
+        // Trigger Transfer
+        $event = new Migration();
+        $event
+            ->setMigration($migration)
+            ->setProject($project)
+            ->setUser($user)
+            ->trigger();
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($migration, Response::MODEL_MIGRATION);
+    });
+
+App::post('/v1/migrations/appwrite/report')
+    ->groups(['api', 'migrations'])
+    ->desc('Generate a report on Appwrite Data')
+    ->label('scope', 'migrations.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'generateAppwriteReport')
+    ->label('sdk.description', '/docs/references/migrations/migration-appwrite-report.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MIGRATION_REPORT)
+    ->param('resources', [], new ArrayList(new WhiteList(Appwrite::getSupportedResources())), 'List of resources to migrate')
+    ->param('endpoint', '', new URL(), "Source's Appwrite Endpoint")
+    ->param('projectID', '', new Text(512), "Source's Project ID")
+    ->param('key', '', new Text(512), "Source's API Key")
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->action(function (array $resources, string $endpoint, string $projectID, string $key, Response $response) {
+
+        $appwrite = new Appwrite($projectID, $endpoint, $key);
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic(new Document($appwrite->report($resources)), Response::MODEL_MIGRATION_REPORT);
+    });
+
+
+App::post('/v1/migrations/firebase')
+    ->groups(['api', 'migrations'])
+    ->desc('Migrate Firebase Data')
+    ->label('scope', 'migrations.write')
+    ->label('event', 'migrations.create')
+    ->label('audits.event', 'migration.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'migrateFirebase')
+    ->label('sdk.description', '/docs/references/migrations/migration-firebase.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MIGRATION)
+    ->param('resources', [], new ArrayList(new WhiteList(Firebase::getSupportedResources())), 'List of resources to migrate')
+    ->param('serviceAccount', '', new Text(512), "Source's Service Account")
```

Comment:
Source's service account sounds vague. This is a firebase specific endpoint so lets make the description firebase specific